sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/models/CertificateProperties.java

```diff
@@ -62,42 +80,75 @@ public class CertificateProperties {
      * include: 'Purgeable', 'Recoverable+Purgeable', 'Recoverable',
      * 'Recoverable+ProtectedSubscription'.
      */
-    private String recoveryLevel;
-
-    /**
-     * The Certificate name.
-     */
-    String name;
+    private final String recoveryLevel;
 
     /**
      * The certificate id.
      */
-    @JsonProperty(value = "id", access = JsonProperty.Access.WRITE_ONLY)
-    private String id;
+    private final String id;
 
     /**
      * Application specific metadata in the form of key-value pairs.
      */
-    @JsonProperty(value = "tags")
-    Map<String, String> tags;
+    private Map<String, String> tags;
 
     /**
      * Thumbprint of the certificate. Read Only
      */
-    @JsonProperty(value = "x5t", access = JsonProperty.Access.WRITE_ONLY)
-    Base64Url x509Thumbprint;
+    private final Base64Url x509Thumbprint;
 
     /**
      * The number of days a certificate is retained before being deleted for a soft delete-enabled Key Vault.
      */
-    @JsonProperty(value = "recoverableDays", access = JsonProperty.Access.WRITE_ONLY)
-    private Integer recoverableDays;
+    private final Integer recoverableDays;
 
-    CertificateProperties(String name) {
-        this.name = name;
+    CertificateProperties() {
```

Comment:
Doesn't need it for serialization, but that doesn't mean we don't need it for usability purposes. Since there were private and package-private constructors it meant there wasn't an implicit ctor.